### ISSUE TITLE
Google data not loading from public document link without login

### ISSUE THREAD

**Issue by lozanaross** [OpenRefine/OpenRefine#5444]
Adding a public document via the Google data tab produces error.

### To Reproduce
Try loading data from a public google document in the latest 3.7 snapshot release.

### Current Results
403 Error (see screenshot below)

### Expected Behavior
Data should be possible to load even without logging in.

### Screenshots
![Screenshot 2022-11-16 at 00 33 43](https://user-images.githubusercontent.com/25489750/202047893-21083037-8d48-4066-99c2-52458fa15893.png)

### Versions
 - Operating System: Apple OS
 - Browser Version: Chrome and Firefox (latest)
 - OpenRefine: tested with 3.7 snapshot release & manual built from master branch

### Additional context
Error first encountered during user testing session on Sept. 30th

**Comment by wetneb** [comment]
Yes, that is expected because an API key is required even for accessing public documents.
We only insert the API key in the official releases, not in the snapshot releases.
When building the tool locally you can follow those instructions to add the required API key:
https://github.com/OpenRefine/OpenRefine/wiki/Google-Extension

We could instead display a better error message indicating that this API key is missing, or simply remove the Google Data extension from the snapshot releases by default.
This extension could also be a good candidate for externalization (removing it from this repository and expecting it to be installed manually by users).

**Comment by thadguidry** [comment]
@wetneb If the Google drive extension was external and installed manually by users, how would things become easier as far as the API key requirement.  I am trying to envision how making the extension manually installed would make it easier for Google drive connections for our users.  Can you elaborate with a step by step in your reply?

**Comment by wetneb** [comment]
If the Google drive extension was external, it could be installed in any version of OpenRefine (built manually from the source, snapshot release or official release) and would come with its own Google API credentials, so OpenRefine itself would not need to store those credentials.

**Comment by tfmorris** [comment]
I'm seeing this with the official 3.7.6 release as well.

**Comment by tfmorris** [comment]
My best guess is that this is a symptom of the application being built/packaged with a valid OAuth clientID & secret, but an invalid API key. The API key isn't needed for OAuth authentication. If the user is authenticated, that authentication token takes priority over the API key. The API key is only needed for unauthenticated access, e.g. access to a public spreadsheet without having the user authenticate.
